### DISCUSSION TITLE
Using strategy.matrix values in the specification of actions to be used in a workflow

### DISCUSSION THREAD

**Discussion by pmoura** [community/community#25824] [Actions]
<p>Hi,</p>
<p>Wondering if it is possible (or is planned to make it possible) to have workflows where strategy.matrix values are used in the specification of actions to be used in a workflow. For example:</p>
<pre><code>name: Testing

on: [push]

jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        backend: [eclipse, swi-prolog]
    steps:
      - name: Install backend
        uses: logtalk-actions/setup-${{ matrix.backend }}@master
        with:
          ${{ matrix.backend }}-version: latest
</code></pre>
<p>I do get an error with the above code:</p>
<pre><code>- Your workflow file was invalid: The pipeline is not valid. .github/workflows/testing_multiple_backends.yml (Line: 14, Col: 15): Unrecognized named-value: 'matrix'. Located at position 1 within expression: matrix.backend
</code></pre>
<p>But wondring if I’m missing some workaround that would allow such the functionality illustrated above.</p>
<p>Thanks,</p>
<p>Paulo</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi Pmoura,</p>
<p>Variable in action name is not supported, please use exact action name with format ‘{org}/{repo}[/path]<span class="mention">@ref</span>’.</p>
<p>As a workaround, you can add ‘if’ expression as a filter.</p>
<p>Code like below:</p>
<pre><code>- if: matrix.backend == 'eclipse'

uses: logtalk-actions/setup-eclipse@master

- if: matrix.backend == 'swi-prolog'

uses: logtalk-actions/setup-swi-prolog@master
</code></pre>
<p>Hope it helps, thanks.</p>
